### ISSUE TITLE
Preload Fonts: improve detection of external domains

### ISSUE THREAD

**Issue by webtrainingwheels** [wp-media/wp-rocket#3276]
The Preload Fonts setting is for locally, or CDN-hosted fonts only. 

However customers are trying really hard to preload external fonts :) 

If you enter: 
`fonts.gstatic.com/s/poppins/v13/pxiByp8kv8JHgFVrLGT9Z1xlFQ.woff2`
This does not get santized correctly and instead of being removed, it becomes: 
`/fonts.gstatic.com/s/poppins/v13/pxiByp8kv8JHgFVrLGT9Z1xlFQ.woff2`

`/fonts.gstatic.com/s/poppins/v13/pxiByp8kv8JHgFVrLGT9Z1xlFQ.woff2` can also be entered directly and it is saved as-is. 

On the front end of the site, this results in: 
`https://www.example.org/fonts.gstatic.com/s/poppins/v13/pxiByp8kv8JHgFVrLGT9Z1xlFQ.woff2`

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

**Comment by ScottTravisHartley** [comment]
Yeah, this is one minor annoyance I ran into quite recently the google fonts cdn is still the most effective means overall for serving optimized font files. I personally pass the text parameter as well to reduce the total size of the transferred woff2 files as well. But preload through wp rocket granted there is a multitude of other ways of preloading fonts I think it should be added to the plugin. 

One thing I want to note though is that preloading of google fonts would require the addition of the crossorigin attribute  -> https://developer.mozilla.org/en-US/docs/Web/HTML/Preloading_content

**Comment by Tabrisrp** [comment]
### Reproduce the issue ✅ 
Reproduced the issue on local test site
### Identify the root cause ✅ 
This is happening because `wp_parse_url()` doesn't process correctly URLs without any scheme or the `//` in front of it. This kind of URLs ends up being considered as a path, and our current code continue in that case.

### Scope a solution ✅ 
A solution is to use the following code, to prepend the scheme in the case the URL doesn't have one, and is also not a relative path.

This needs to be added to the `Engine/Admin/Settings/Settings/sanitize_font()` method

```
		if (
			'//' !== substr( $file, 0, 2 )
			&&
			'/' !== substr( $file, 0, 1 )
			&&
			'http' !== substr( $file, 0, 4 )
		) {
			$file = rocket_add_url_protocol( $file );
		}
```

A new test case needs to be added to the corresponding fixtures to validate the change and prevent regression.

### Estimate the effort ✅
Effort `[XS]`
